services/app-web/src/components/ExpandableText/ExpandableText.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useLayoutEffect, useState } from 'react'
+import { Button } from '@trussworks/react-uswds'
+import styles from './ExpandableText.module.scss'
+
+export type ExpandableTextProps = {
+    clamp: React.ReactElement | string // you can pass in an element with nested <span> or simple string
+    clampedLines?: number
+}
+
+export const ExpandableText = ({
+    clamp,
+    clampedLines = 2,
+    ...props
+}: ExpandableTextProps &
+    React.HTMLAttributes<HTMLDivElement>): React.ReactElement => {
+    //We are using React.useRef because jest spyOn useRef only works like this.
+    const textRef = React.useRef<HTMLDivElement>(null)
+    const [showMore, setShowMore] = useState<boolean>(false)
+    const [showMoreButton, setShowMoreButton] = useState<boolean>(false)
+
+    useLayoutEffect(() => {
+        const ref = textRef?.current
+        if (
+            ref &&
+            (ref.offsetHeight < ref.scrollHeight ||
+                ref.offsetWidth < ref.scrollWidth)
+        ) {
+            if (clampedLines) {
+                ref.style.webkitLineClamp = clampedLines.toString()
+            }
+            setShowMoreButton(true)
+        }
+    }, [showMoreButton, clampedLines, setShowMoreButton])
+
+    return (
+        <p {...props}>
+            <span
+                ref={textRef}
+                data-testid="clampElement"
```

Comment:
> In regards to the how to Actions tab on storybook is flooded with actions. I've noticed storybook is very slow, almost crashing, when I view this story. I wonder if we have a memory leak in the React state. Going to look more at things here on Monday. I don't want to merge this work if there's some kind of performance issue going on that will make Storybook hard to use.

It's the spreading of props `{...props}` that's causing this issue. I don't think it's needed here or in the Updated and Unlocked banners. If there is no objections I can remove that

Reply:
Huh, yeah that's fine to remove we aren't even passing anything in there lol